compiler/AST/ForallStmt.cpp

```diff
@@ -603,3 +603,44 @@ bool ForallStmt::hasVectorizationHazard() const {
 void ForallStmt::setHasVectorizationHazard(bool v) {
   fVectorizationHazard = v;
 }
+
+static void gatherFollowerLoopBodies(BlockStmt* block,
```

Comment:
This seems similar to the logic used to mark serial yielding loops in followers/standalone iterators as order independent in `resolveFunction.cpp::markIterator()`

Can we use this logic in place of markIterator or do they have to be different since they're used in different points in compilation (they could also be identifying different loops, I'm not positive I'm following this logic)

Reply:
The logic is pretty different since this part finds ast at the location of the `forall` in the source code but `markIterator` is adjusting an iterator itself. I'd like to leave markIterator / orderIndependentSet alone for this PR, but arguably we could unify the representation to the loop-optimization-info.

Reply:
The more I think about it, the more I think that this style of representation, with a loop-optimization-info marker at the end of the follow loop, would be more robust for more analysis, and more gracefully handles the various cases we need to check for as vectorization hazards, so I think we should look into unifying them. Just not in this PR.